arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp

```diff
@@ -132,21 +126,16 @@ void RocksDBRestReplicationHandler::handleCommandBatch() {
 
     // add client
     bool found;
-    std::string const& value = _request->value("serverId", found);
-    if (!found) {
-      LOG_TOPIC("61e69", DEBUG, Logger::REPLICATION)
-          << "no serverId parameter found in request to " << _request->fullUrl();
-    }
-
-    TRI_server_id_t serverId = id;  // just use context id as fallback
-    if (!value.empty() && value != "none") {
-      serverId = static_cast<TRI_server_id_t>(StringUtils::uint64(value));
+    std::string clientId = _request->value("serverId", found);
+    if (clientId.empty() || clientId == "none") {
+      // use context id as a fallback
+      clientId = std::to_string(id);
```

Comment:
I thini the you also implement this in the ReplicationContext above ?

Reply:
Ok.